### DISCUSSION TITLE
Permit use of useRecentTransactions

### DISCUSSION THREAD

**Discussion by exception** [rainbow-me/rainbowkit#373] [Ideas]
Since we already get access to adding a new transaction, would be cool to export the `useRecentTransactions` hook. Would allow users to have more liberty with their UX.

**Comment by 0hq** [comment]
Bump.

**Reply by santiarr** [reply]
Bump

**Comment by spennyp** [comment]
Bump, this would be quite useful!

**Reply by spennyp** [reply]
Opened a PR to add this here: https://github.com/rainbow-me/rainbowkit/pull/675

**Comment by DanielSinclair** [comment]
Curious to hear Transaction-related feature requests. Exposing `useRecentTransactions` might be limiting, whereas a better Transactions API that pushes devs to handle transaction state and transaction history would be more usable (as opposed to the limited scope of RainbowKit-origin transactions).

**Comment by shrugs** [comment]
prompted by the closing of #675 i took a re-look at my codebase and realized i didn't actually need the full functionality of the RainbowTransactionStore, and instead I actually just wanted to dual-publish to my own list of tx hashes and use wagmi's `useWaitForTransaction` to retrieve tx status within the hook that actually cared about it. obviously this pattern results in similar/identical output but doesn't require duplicating or exporting the RainbowTransactionStore

**Comment by 0xbuidler** [comment]
I digged into the code and saw a `useTransactionStore` which returns an object containing a `getTransactions` function. This seems to be **exactly** what we need. I'm just wondering what is the bottleneck here and why this hasn't already been implemented.

**Reply by DanielSinclair** [reply]
@sshmaxime What are you hoping to accomplish? Can have another look. In most cases, using the raw Wagmi hooks within your dApp will be a better path.

**Reply by 0xbuidler** [reply]
Hey @DanielSinclair , thank you so much for the fast reply and what you guys are building.

Well, I want to implement a loader (already doable as of now with `account.hasPendingTransactions`) but on hover I would need it to display a dropdown with the list of the pending transactions (their `hash` and `description` at least).

**Reply by 0xbuidler** [reply]
Or there should be a way to specify a link or something on the account modal. It's frustrating because the main functionalities are there but it misses maybe the most crucial part which is redirecting the user to an `url` or something when viewing the `Recent Transactions` for each txs. Like a link to Etherscan or something in this spirit.
